File: sum-natural-numbers.js
const sumNaturalNumbers = (myInt, myMultiple) => {

    // ended up movinig this to its own function since the logic us used twice
    const isMultiple = (i, value) => {
        return ((i % value) === 0);
    };

    const processArray = (i) => {
        // using every instead of map and returning false to break the loop after finding a match
        // otherwise 15 (which is a multiple of 3 and 5) would be added twice
        myMultiple.every(value => {
            if (isMultiple(i, value)) {
                myReturn += i;
                return false;
            } else {
                return true;
            }
        });
    };

    let i;
    let myReturn = 0;
    let myType = typeof(myMultiple);

    for (i = 0; i <= myInt; i++) {
        if (myType === 'object') {
            if (myMultiple.length) {
                processArray(i);
            } else {
                return 'TODO: handle objects';
            }
        }
        if (myType === 'number') {
            if (isMultiple(i, myMultiple)) {
                myReturn += i;
            }
        }
        if (myType === 'undefined') {
            myReturn += i;
        }
    }
    return myReturn;
};


// some tests
console.log('test 1: ', sumNaturalNumbers(100, [3, 5, 11]));
console.log('test 2: ', sumNaturalNumbers(10));
console.log('test 3: ', sumNaturalNumbers(12, 3));

File: sum_natural_numbers.py
def sum_natural_numbers(myInt, myMultiple=""):

    def is_multiple(i, myMultiple):
        return ((i % myMultiple) == 0)

    def process_array(i, myMultiple):
        myValue = 0
        for n in myMultiple:
            if is_multiple(i, n):
                myValue = i
                break
        return myValue

    i = 0
    myReturn = 0
    myType = type(myMultiple)

    for i in range(myInt + 1):
        if myType is type(""):
            myReturn += i
        if myType is type(1):
            if(is_multiple(i, myMultiple)):
                myReturn += i
        if myType is type([]):
            myReturn += process_array(i, myMultiple)

    return myReturn

if __name__ == '__main__':

    # some tests
    print('test 1: ',  sum_natural_numbers(10))
    print('test 2: ', sum_natural_numbers(12, 3))
    print('test 3: ', sum_natural_numbers(100, [3, 5, 11]))